### ISSUE TITLE
Request for fix: False positive AvoidSpringMVCMemoryLeak

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#33]
This code gives a false positive, please fix:

```java
@RequestMapping(value = "/info/all", method = RequestMethod.GET)
public ResponseEntity agentInfoAll() {
    String result = getAllServers()
            .stream()
            .filter(server -> !server.startsWith(GROUP_OF_SERVERS_PREFIX))
            .map(server -> {
                CompletableFuture<ResponseEntity<String>> futureResponse =
                        CompletableFuture.supplyAsync(() -> agentClient.pingAgent(server), executor);
                ResponseEntity<String> response = futureResponse
                        .handle((res, ex) -> (res != null) ? res : ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error pinging server: " + ex.getMessage()))
                        .join();
                return server + " : " + buildStatus(response) + "\n" + response.getBody() + "\n\n";
            })
            .collect(Collectors.joining());

    return ResponseEntity.ok().contentType(MediaType.TEXT_PLAIN).body(Pattern.compile("\n{2,}").matcher(result).replaceAll("\n\n"));
}
```

**Comment by jborgers** [comment]
Does it still add value to fix this?
